src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -113,8 +113,59 @@ void
 PeerImp::run()
 {
     if(! strand_.running_in_this_thread())
-        return post(strand_, std::bind (
-            &PeerImp::run, shared_from_this()));
+        return post(strand_, std::bind(&PeerImp::run, shared_from_this()));
+
+    // We need to decipher
+    auto parseLedgerHash = [](std::string const& value) -> boost::optional<uint256>
+    {
+        uint256 ret;
+        if (ret.SetHexExact(value))
+            return { ret };
+
+        auto const s = base64_decode(value);
+        if (s.size() != uint256::size())
+            return boost::none;
+        return uint256{ s };
+    };
+
+    boost::optional<uint256> closed;
+    boost::optional<uint256> previous;
+
+    {
+        auto const iter = headers_.find("Closed-Ledger");
+
+        if (iter != headers_.end())
+        {
+            closed = parseLedgerHash(iter->value().to_string());
+
+            if (!closed)
+                fail("Malformed handshake data (1)");
+        }
+    }
+
+    {
+        auto const iter = headers_.find("Previous-Ledger");
+
+        if (iter != headers_.end())
+        {
+            previous = parseLedgerHash(iter->value().to_string());
+
+            if (!previous)
+                fail("Malformed handshake data (2)");
+        }
+    }
+
+    if (previous && !closed)
+        fail("Malformed handshake data (3)");
+
+    {
+        std::lock_guard<std::mutex> sl(recentLock_);
```

Comment:
Just `std::lock_guard` works (class template argument deduction works well with `lock_guard`)

Reply:
:open_mouth: I didn't know that.